app/services/clients/utils/generateUrl.js

```diff
@@ -0,0 +1,30 @@
+function generateUrl (template, urlParams = {}, queryParams = {}) {
```

Comment:
I take it it doesn't matter what the path params are in the template? For e.g. `/v1/{example}` would be treated the same way as `/v1/{gatewayId}`?
Also although there are comprehensive tests for this function I'm finding it really hard to understand how it works.

Reply:
Can we use's [axios to encode query params](https://axios-http.com/docs/urlencoded)? Then for the normal url we can revert to 
```
`${this.connectorUrl}/v1/api/accounts/{accountId}/stripe-account`
.replace('{accountId}', encodeURIComponent(gatewayAccountId))
```

Reply:
You can choose any name for the template variables and they all need to be provided when using the template.

I think it's good to put the full URL generation together in one place rather than splitting some of it out to Axios.

The Axios implementation also does some quite specific things with arrays that I wouldn't expect to be compatible with our Java backends - they remind me of PHP's handling of query strings.

Even if we want to use Axios's approach I'd build it into the generator so it's a centralised change if we want to change it.